### PR TITLE
Implemented CPT-28: Drop down CMS collection

### PR DESCRIPTION
**Pull request by maxim-babenkocommerce**

### CODE REVIEW

File: Ui/Component/Form/Store/StoreViews.php [hunk 1]
```diff
@@ -21,14 +39,26 @@ public function toOptionArray()
                 $this->sanitizeName($store->getGroup()->getName()). ' | ' .
                 $this->sanitizeName($store->getName());
             $storeId = $store->getId();
+            $cmsPageOptions = [[
+                'label' => __('-- No Linked Page --'),
+                'value' => '0',
+            ]];
+            $cmsPageCollection = $this->cmsPageCollectionFactory->create();
+            foreach ($cmsPageCollection as $cmsPage) {
```
Comment (vsevolod-tarasov) [inline Ui/Component/Form/Store/StoreViews.php]: - why don`t we use addFieldToFilter?

Comment (vsevolod-tarasov) [reply Ui/Component/Form/Store/StoreViews.php]: $this->pageCollection->addFilter('store', ['in' => [1,2]], 'public')->load();

File: Ui/Component/Form/Store/StoreViews.php [hunk 2]
```diff
@@ -13,22 +31,39 @@ class StoreViews extends StoreOptions
     public function toOptionArray()
     {
         $result = [];
+        $stores = [];
         $i=0;
         $storeCollection = $this->systemStore->getStoreCollection();
+        foreach ($storeCollection as $store) {
+            $stores[] = $store->getId();
```
Comment (vsevolod-tarasov) [inline Ui/Component/Form/Store/StoreViews.php]: $stores[]  = array_keys($storeCollection);

File: Ui/Component/Form/Store/StoreViews.php [hunk 3]
```diff
@@ -13,22 +31,39 @@ class StoreViews extends StoreOptions
...
+        }
+        $cmsPageCollection = $this->cmsPageCollectionFactory->create()
+            ->addFilter('store', ['in' => $stores], 'public')->load();
         /** @var  \Magento\Store\Model\Store $store */
         foreach ($storeCollection as $store) {
```
Comment (vsevolod-tarasov) [inline Ui/Component/Form/Store/StoreViews.php]: foreach ($storeCollection as $storeId => $store) {

File: Ui/Component/Form/Store/StoreViews.php [hunk 4]
```diff
@@ -13,22 +31,39 @@ class StoreViews extends StoreOptions
...
+                $storeIds = $cmsPage->getStoreId();
+                if ((count($storeIds) != count($storeCollection)) and (in_array($storeId, $storeIds))){
+                    $cmsPageOptions[] = [
+                        'label' => $cmsPage->getTitle(),
+                        'value' => $cmsPage->getId(),
+                    ];
+                }
+            }
             $result[$i] = [
                 'record_id' => $i,
```
Comment (vsevolod-tarasov) [inline Ui/Component/Form/Store/StoreViews.php]: $result[$storeId] = [
                'record_id' => $storeId,
                'store_id' => $storeId,
                'store_label' => $storeLabel,
                'linked_page_id' => '0',
                'cms_page_options' => $cmsPageOptions,
            ];

File: Ui/Component/Form/Store/StoreViews.php [hunk 5]
```diff
@@ -13,22 +31,39 @@ class StoreViews extends StoreOptions
     public function toOptionArray()
     {
         $result = [];
+        $stores = [];
```
Comment (vsevolod-tarasov) [inline Ui/Component/Form/Store/StoreViews.php]: if apply all of below, the $stores and $i could be removed

File: Ui/Component/Form/Store/StoreViews.php [hunk 6]
```diff
@@ -13,22 +31,39 @@ class StoreViews extends StoreOptions
     public function toOptionArray()
```
Comment (vsevolod-tarasov) [inline Ui/Component/Form/Store/StoreViews.php]: there are lot of re-factoring for code please look closely

File: Ui/Component/Form/Store/StoreViews.php (hunk 4 above)
Comment (maxim-babenkocommerce) [reply Ui/Component/Form/Store/StoreViews.php]: we will receive incorrect for UI array

File: Ui/Component/Form/Store/StoreViews.php (hunk 3 above)
Comment (maxim-babenkocommerce) [reply Ui/Component/Form/Store/StoreViews.php]: such $storeId will return an integer value, we need a string

**Pull request merged**